### PR TITLE
Add pagination to observations (and split executed and queued)

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds pagination to the observation page (to start partially addressing https://github.com/skyportal/skyportal/issues/2885). It also splits the concept of Executed and Queued observations which should be handled differently.

### CODE REVIEW

File: static/js/ducks/observations.js [hunk 1]
```diff
@@ -43,6 +45,10 @@ export function fetchObservations(filterParams = {}) {
     filterParams.endDate = dayjs().utc().format("YYYY-MM-DDTHH:mm:ssZ");
   }
 
+  if (!Object.keys(filterParams).includes("numPerPage")) {
+    filterParams.numPerPage = 10;
```
Comment (guynir42) [inline static/js/ducks/observations.js:49]: Please make sure the default is not defined in more places than it should be (i.e., once for API calls, once for front end, at the most).

File: static/js/ducks/queued_observations.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  if (!Object.keys(filterParams).includes("endDate")) {
+    if (dateobs) {
+      filterParams.endDate = dayjs(dateobs)
+        .add(7, "day")
+        .format("YYYY-MM-DD HH:mm:ss");
+    }
+  }
+
+  if (!Object.keys(filterParams).includes("numPerPage")) {
+    filterParams.numPerPage = 10;
```
Comment (guynir42) [inline static/js/ducks/queued_observations.js:74]: Same here for the default.

**Review by guynir42 — APPROVED** [review]
Most of my comments, as usual, regard the documentation.

File: skyportal/facility_apis/ztf.py [hunk 3]
```diff
@@ -718,6 +719,42 @@ def remove(request):
...
+        """
+
+        altdata = allocation.altdata
+        if not altdata:
+            raise ValueError('Missing allocation information.')
+
+        headers = {"Authorization": f"Bearer {altdata['access_token']}"}
+
+        url = urllib.parse.urljoin(ZTF_URL, 'api/triggers/ztf')
+        s = Session()
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/facility_apis/ztf.py:740]: This is actually an SSH session, just to be extra confusing ;)

File: skyportal/facility_apis/ztf.py [hunk 4]
```diff
@@ -821,3 +858,58 @@ def fetch_observations(instrument_id, client, request_str):
     from skyportal.handlers.api.observation import add_observations
 
     add_observations(instrument_id, obstable)
+
+
+def fetch_queued_observations(instrument_id, obstable):
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/facility_apis/ztf.py:863]: This is what is triggered in the async part to actually grab the observations.

File: skyportal/handlers/api/observation.py [hunk 5]
```diff
@@ -43,6 +44,64 @@
 
 Session = scoped_session(sessionmaker(bind=DBSession.session_factory.kw["bind"]))
 
+MAX_OBSERVATIONS = 1000
+
+
+def add_queued_observations(instrument_id, obstable):
+    """Fetch queued observations from ZTF scheduler.
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/observation.py:51]: yup.

File: skyportal/handlers/api/observation.py [hunk 6]
```diff
@@ -618,6 +694,27 @@ def get(self):
...
+              description: |
+                 Whether to include queued or executed observations.
+                 Defaults to executed.
+            - in: query
+              name: numPerPage
+              nullable: true
+              schema:
+                type: integer
+              description: |
+                Number of followup requests to return per paginated request. Defaults to 100. Can be no larger than {MAX_OBSERVATIONS}.
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/observation.py:711]: of course.

File: skyportal/handlers/api/observation.py [hunk 7]
```diff
@@ -618,6 +694,27 @@ def get(self):
               description: |
                 Boolean indicating whether to include associated GeoJSON. Defaults to
                 false.
+            - in: query
+              name: observationType
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply skyportal/handlers/api/observation.py:698]: yeah thats better

File: static/js/ducks/observations.js (hunk 1 above)
Comment (mcoughlin) [reply static/js/ducks/observations.js:49]: I think this duck is the right place for this one.

File: static/js/ducks/queued_observations.js (hunk 2 above)
Comment (mcoughlin) [reply static/js/ducks/queued_observations.js:74]: This one too.

**Pull request merged**